ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryLimitFunctionalTest.java

```diff
@@ -0,0 +1,491 @@
+import io.confluent.ksql.rest.server.utils.TestUtils;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import io.confluent.ksql.serde.FormatFactory;
+import io.confluent.ksql.serde.SerdeFeatures;
+import io.confluent.ksql.test.util.KsqlIdentifierTestUtil;
+import io.confluent.ksql.test.util.KsqlTestFolder;
+import io.confluent.ksql.test.util.TestBasicJaasConfig;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.UserDataProviderBig;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import kafka.zookeeper.ZooKeeperClientException;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.streams.StreamsConfig;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.junit.rules.Timeout;
+
+
+@Category({IntegrationTest.class})
+public class PullQueryLimitFunctionalTest {
+
+    private static final String USER_TOPIC = "user_topic_";
+    private static final String USERS_STREAM = "users";
+    private static final UserDataProviderBig USER_PROVIDER = new UserDataProviderBig();
+    private static final int TOTAL_RECORDS = USER_PROVIDER.getNumRecords();
+    private static final int HEADER = 1;
+    private static final int LIMIT_REACHED_MESSAGE = 1;
+    private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+    private static final TemporaryFolder TMP = KsqlTestFolder.temporaryFolder();
+    private static final int BASE_TIME = 1_000_000;
+    private final AtomicLong timestampSupplier = new AtomicLong(BASE_TIME);
+    private String output;
+    private String queryId;
+    private String topic;
+
+    private static final PhysicalSchema AGGREGATE_SCHEMA = PhysicalSchema.from(
+            LogicalSchema.builder()
+                    .keyColumn(ColumnName.of("USERID"), SqlTypes.STRING)
+                    .valueColumn(ColumnName.of("COUNT"), SqlTypes.BIGINT)
+                    .build(),
+            SerdeFeatures.of(),
+            SerdeFeatures.of()
+    );
+
+    private static final String PROPS_JAAS_REALM = "KsqlServer-Props";
+    private static final String KSQL_RESOURCE = "ksql-user";
+    private static final String USER_WITH_ACCESS = "harry";
+    private static final String USER_WITH_ACCESS_PWD = "changeme";
+    private static final Optional<BasicCredentials> USER_CREDS
+            = Optional.of(BasicCredentials.of(USER_WITH_ACCESS, USER_WITH_ACCESS_PWD));
+
+    @ClassRule
+    public static final TestBasicJaasConfig JAAS_CONFIG = TestBasicJaasConfig
```

Comment:
It's fine to include auth in this test, though not entirely necessary since we have pull query tests with it elsewhere and you could simplify this.

Reply:
removed